src/Services/LightningService.cs

```diff
@@ -635,6 +562,96 @@ await foreach (var response in openStatusUpdateStream.ResponseStream.ReadAllAsyn
             return new Lightning.LightningClient(grpcChannel).Wrap();
         };
 
+        public async Task OnStatusChannelOpened(ChannelOperationRequest channelOperationRequest, Node source, ChannelPoint channelPoint, string? closeAddress = null)
+        {
+            await using var context = await _dbContextFactory.CreateDbContextAsync();
+
+            _logger.LogInformation(
+                "Channel opened for channel operation request request id: {RequestId}, channel point: {ChannelPoint}",
+                channelOperationRequest.Id, channelPoint.ToString());
+
+            channelOperationRequest.Status = ChannelOperationRequestStatus.OnChainConfirmed;
+            if (channelOperationRequest.StatusLogs.Count > 0)
+            {
+                channelOperationRequest.StatusLogs.Add(ChannelStatusLog.Info($"Channel opened successfully 🎉"));
+            }
+
+            var (isSuccess, error) = _channelOperationRequestRepository.Update(channelOperationRequest);
+            if (!isSuccess)
+            {
+                _logger.LogWarning("Request is in OnChainConfirmed, but could not update status for request id: {RequestId} reason: {Reason}", channelOperationRequest.Id, error);
+            }
+
+            var channel = await CreateChannel(source, channelOperationRequest.DestNode.Id, channelPoint, channelOperationRequest.SatsAmount, closeAddress);
+
+            var channelExists = await _channelRepository.GetByChanId(channel.ChanId);
+            if (channelExists == null)
+                await context.AddAsync(channel);
+            else
+            {
+                channel.Id = channelExists.Id;
+                context.Update(channel);
+            }
+
+            var addChannelResult = (await context.SaveChangesAsync()) > 0;
+
+            if (addChannelResult == false)
+            {
+                _logger.LogError(
+                    "Channel for channel operation request id: {RequestId} could not be created, reason: {Reason}",
+                    channelOperationRequest.Id,
+                    "Could not persist to db");
+            }
+
+            channelOperationRequest.ChannelId = channel?.Id;
+            channelOperationRequest.DestNode = null;
+
+            var channelUpdate = _channelOperationRequestRepository.Update(channelOperationRequest);
+
+            if (channelUpdate.Item1 == false)
+            {
+                _logger.LogError(
+                    "Could not assign channel id to channel operation request: {RequestId} reason: {Reason}",
+                    channelOperationRequest.Id,
+                    channelUpdate.Item2);
+            }
+        }
+
+        public static async Task<Channel> CreateChannel(Node source, int destId, ChannelPoint channelPoint, long satsAmount, string? closeAddress = null)
```

Comment:
Is this the best place to put this logic? 🤔  just thinking

Reply:
It was there already, I just moved it to it's own function. I agree that this method is insanely large, so we should start thinking about moving stuff